modules/modules_test.go

```diff
@@ -377,16 +377,17 @@ func TestModuleService_InterruptedFastStartup(t *testing.T) {
 	go func() {
 		<-subserviceStarted
 		cancel()
+		close(finishStarting)
 	}()
 
-	require.ErrorIs(t, services.StartAndAwaitRunning(ctx, moduleSvc), context.Canceled)
+	// Start service using context that's going to be cancelled
+	require.NoError(t, moduleSvc.StartAsync(ctx))
+	// We get context cancelled error from service context cancellation.
+	require.ErrorIs(t, moduleSvc.AwaitRunning(context.Background()), context.Canceled)
```

Comment:
Note that original test used `ctx` for waiting, but that wait got canceled in the test.